src/response.ts

```diff
@@ -279,6 +279,13 @@ export class Response extends Macroable {
       this.safeStatus(204)
     }
 
+    /**
+     *
+     */
+    if (this.#config.generateRequestId) {
+      this.header('X-Request-Id', this.request.headers['x-request-id'])
```

Comment:
I think the `X-Request-Id` should be set if its available in the request headers, regardless of whether it is enabled within the AdonisJS app or not.

This is because, the frontend might set the header, the backend app never generates it and re-uses the existing header and in that case, we still want the id to exist in the response.